lib/game_collection.rb

```diff
@@ -94,4 +93,49 @@ def biggest_blowout
     end.difference_between_score
     games_difference
   end
+
+  
+
+  def find_away_defense_goals(away_team_id)
+    away_defense = @games.find_all {|game| game.away_team_id == (away_team_id)}
+
+    away_defense.map {|game| game.home_goals}
+  end
+
+  def find_home_defense_goals(home_team_id)
+    home_defense = @games.find_all {|game| game.home_team_id == (home_team_id)}
+
+    home_defense.map {|game| game.away_goals}
+  end
+
+  def find_average_defense_goals(team_id)
+    defense_goals_array = find_home_defense_goals(team_id) + find_away_defense_goals(team_id)
+
+    goals_total = defense_goals_array.reduce(0) {|sum, defense_score| sum + defense_score}
+
+    average = (goals_total.to_f / defense_goals_array.length).round(2)
+  end
```

Comment:
we reuse average quite a bit. could we incorporate something to reduce code come refactor time?

Reply:
I believe the average goals method refers to the collection of all games. With this method/logic it needs to be team specific.

Reply:
potential for a module down the road